src/main/java/net/minecraftforge/client/model/pipeline/BlockInfo.java

```diff
@@ -158,6 +162,18 @@ public void updateLightMatrix()
         }
     }
 
+    public void updateFlatLighting()
```

Comment:
I recommend you combine the data from this (as you'll need a 3x3x3 cube of data, see my other comment) with this one: https://github.com/bs2609/MinecraftForge/blob/86e4317d1255eeebf848516364c391aa457a5093/src/main/java/net/minecraftforge/client/model/pipeline/BlockInfo.java#L117

Reply:
I'm keeping the data seperate as it's handled differently, and the added flat lighting array is small compared to the other data arrays. I can understand the desire to not have duplicated data, but I think this could potentially be better achieved in future by having seperate `BlockInfo` implementations.